container/lxd/client_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package lxd_test
+
+import (
+	"github.com/golang/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/container/lxd"
+	lxdtesting "github.com/juju/juju/container/lxd/testing"
+	coretesting "github.com/juju/juju/testing"
+	"github.com/lxc/lxd/shared/api"
+)
+
+type clientSuite struct {
+	coretesting.BaseSuite
+}
+
+var _ = gc.Suite(&clientSuite{})
+
+const eTag = "etag"
+
+func (s *connectionSuite) TestUpdateServerConfig(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+	cSvr := lxdtesting.NewMockContainerServer(ctrl)
+
+	newConfig := map[string]interface{}{"key1": "val1"}
+	updateReq := api.ServerPut{Config: newConfig}
+	gomock.InOrder(
+		cSvr.EXPECT().GetServer().Return(&api.Server{}, eTag, nil),
+		cSvr.EXPECT().UpdateServer(updateReq, eTag).Return(nil),
```

Comment:
I'm still a little unclear on what gomock will assert with respect to this.

Does it assert that the args to UpdateServer match?

Reply:
Yes.